src/auth.ts

```diff
@@ -66,21 +66,22 @@ async function authorize(useLocalhost: boolean, writeToOwnKey: boolean) {
 /**
  * Loads the Apps Script API credentials for the CLI.
  * Required before every API call.
- * @param {Function} cb The callback
- * @param {boolean} isLocal If we should load local API credentials for this clasp project.
  */
-export function getAPICredentials(cb: (rc: ClaspSettings | void) => void) {
-    DOTFILE.RC_LOCAL.read().then((rc: ClaspSettings) => {
-      oauth2Client.setCredentials(rc);
-      cb(rc);
-    }).catch((err: any) => {
-      DOTFILE.RC.read().then((rc: ClaspSettings) => {
+export async function loadAPICredentials() {
+    return new Promise((resolve, reject) => {
```

Comment:
Is this how we want it? Tests pass so it seems to work, just want to make sure this is the best way forward.

Reply:
No. `DOTFILE.RC_LOCAL.read()` already returns a Promise. There's no need to wrap a Promise around the Promise. Promises can be chained too, so we're okay to leave this as is. All we care about is loading `oauth2Client` before calling API methods, it really doesn't matter what happens after that.

You can try this in Chrome console:
```js
var a = new Promise();
function c() {
    return a.then(() => {}).catch(() => {});
}
c();
// Promise {<pending>}
```